### PR TITLE
Assorted cleanups of stable runtime crates

### PR DESCRIPTION
**Pull request by rcoh**
## Motivation and Context
Stable crates MUST only expose other stable crates.

## Description
 This:
- fixes the remaining issues
- adds a lint tool to be sure we don't expose unstable crates by accident in the future

## Testing
CI run

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: buildSrc/src/test/kotlin/CrateSetTest.kt [hunk 1]
```diff
@@ -41,17 +41,16 @@ class CrateSetTest {
         crateSet.forEach {
             val path = "$relativePathToRustRuntime/${it.name}/Cargo.toml"
             val contents = File(path).readText()
-            val manifest = try {
-                Toml().read(contents)
+            if (it.name == "aws-sigv4") {
+                print(contents)
+            }
```
Comment (jdisanti) [inline buildSrc/src/test/kotlin/CrateSetTest.kt:46]: Looks like a debug print? Did you intend to keep this?

File: buildSrc/src/test/kotlin/CrateSetTest.kt [hunk 2]
```diff
@@ -41,17 +41,16 @@ class CrateSetTest {
...
+            val isStable = try {
+                Toml().read(contents).getTable("package.metadata.smithy-rs-release-tooling")?.getBoolean("stable") ?: false
             } catch (e: java.lang.IllegalStateException) {
-                // Currently, `aws-sigv4` cannot be read as a `TOML` because of the following error:
-                // Invalid table definition on line 54: [target.'cfg(not(any(target_arch = "powerpc", target_arch = "powerpc64")))'.dev-dependencies]]
-                logger.info("failed to read ${it.name} as TOML: $e")
-                Toml()
+                // sigv4 doesn't parse but it's stable now, hax hax hax
```
Comment (jdisanti) [inline buildSrc/src/test/kotlin/CrateSetTest.kt:50]: 😆

File: tools/ci-build/sdk-lints/src/lint_cargo_toml.rs [hunk 3]
```diff
@@ -224,3 +227,170 @@ fn fix_docs_rs(contents: &str) -> Result<String> {
...
+    #[serde(rename = "smithy-rs-release-tooling")]
+    smithy_rs_release_tooling: ToolingMetadata,
+}
+
+#[derive(Deserialize)]
+struct ToolingMetadata {
+    stable: bool,
+}
+
+pub(crate) struct StableCratesExposeStableCrates {
```
Comment (jdisanti) [inline tools/ci-build/sdk-lints/src/lint_cargo_toml.rs:242]: Neat! Check external types inception

**Review by jdisanti — APPROVED** [review]

**Comment by rcoh** [comment]
No will remove

File: tools/ci-build/sdk-lints/src/lint_cargo_toml.rs
Comment (ysaito1001) [inline tools/ci-build/sdk-lints/src/lint_cargo_toml.rs:1]: Love this lint ❤️

**Review by ysaito1001 — APPROVED** [review]

**Pull request merged**